src/main/java/com/amazon/opendistroforelasticsearch/sql/antlr/visitor/AntlrSqlParseTreeVisitor.java

```diff
@@ -174,6 +175,11 @@ public T visitQuerySpecification(QuerySpecificationContext ctx) {
         return result;
     }
 
+    @Override
+    public T visitSubqueryTableItem(SubqueryTableItemContext ctx) {
+        throw new EarlyExitAnalysisException("Exit when meeting subquery in from");
```

Comment:
Please help open an enhancement issue so we can try to figure out later. Thanks!

Reply:
Done. https://github.com/opendistro-for-elasticsearch/sql/issues/281